### PR TITLE
Feature/backend chat hendrik

### PR DESCRIPTION
**Pull request by Hevhove**
for Eduardo:

Endpoint:

PATCH /chat/channels/:channelId

This endpoint allows the owner of a chat channel to edit the channel.

Roles:

Owner of the channel.

Request body (EditChannelDto):

```
{
    "channelType": "<New Channel Type>",
    "name": "<New Channel Name>",
    "avatar": "<New Avatar>",
    "password": "<New Password>"
}

```

Details:

    channelId: This is a path parameter. Replace :channelId with the ID of the channel you want to edit.

    channelType: This field is optional. It's the type of the channel you're editing. The value should be either "PUBLIC", "PRIVATE", "PROTECTED", or "DM". If you don't provide this, the server will use the current channel type.

    name: This field is optional. It's the new name you want to give to the channel. The server will throw an error if the provided name is the same as the current name.

    avatar: This field is optional. It's the new avatar you want to assign to the channel.

    password: This field is optional. It's the new password you want to set for the channel. If you're setting or changing the password, the channel type must be "PROTECTED". If you're removing the password, the channel type must be "PUBLIC", and you should leave this field empty.
    
    
   Notes:

    A channel with the type "DM" cannot be edited.

    If you want to remove the password, the channel type should be "PUBLIC", and you should leave the password field empty.

    If you want to change the password, the channel type should be "PROTECTED", and you should provide a new password in the password field.

    If you want to add a password to a public channel, you should set channelType to "PROTECTED" and provide a new password.

### CODE REVIEW

**Review by luis-prates — CHANGES_REQUESTED** [review]
Looks good! Please pull the latest dev changes into this branch, resolve any conflicts that may arise, and run the lint and format commands, please. Don't know if you have the `npm run lint:format` command set up.

File: backend/src/types/channel-type.enum.ts
[the remark this answers is not in the document]
Comment (Hevhove) [reply backend/src/types/channel-type.enum.ts:1]: It's supposed to be deleted. I made this myself but actually, this enum is also defined inside of the prisma client. So I was using 2 different enums with the same function. Eveything was harmonized to the enum `channelType` from @prisma/client now.

File: backend/src/chat/chat.controller.ts [hunk 1]
```diff
@@ -43,6 +43,14 @@ export class ChatController {
 		return this.chatService.createChannel(createChannelDto, req.user);
 	}
 
+    // Edit a channel
+    @Patch(':channelId')
```
[the remark this answers is not in the document]
Comment (Hevhove) [reply backend/src/chat/chat.controller.ts:47]: Ok, I will change the path to `edit/:channelId`

**Review by luis-prates — APPROVED** [review]
Good job and thank you for changing based on the suggestions I made!

**Pull request merged**